tooling/greybox_fuzzer/src/lib.rs

```diff
@@ -996,22 +992,28 @@ impl<
         match (result_acir, result_brillig) {
             (Ok(witnesses), Ok((_map, brillig_coverage))) => {
                 // If we expect the program to always fail, we need to return a counter example
-                if self.failure_configuration.should_always_fail {
-                    return HarnessExecutionOutcome::CounterExample(CounterExampleOutcome {
-                        case_id: testcase.id(),
-                        exit_reason: "".to_string(),
-                        counterexample: testcase.value().clone(),
-                    });
+                match self.failure_configuration {
+                    // If we expect the program to always fail, we need to return a counter example, if it actually passes
+                    FuzzedExecutorFailureConfiguration::ShouldFail
+                    | FuzzedExecutorFailureConfiguration::ShouldFailWith(_) => {
+                        HarnessExecutionOutcome::CounterExample(CounterExampleOutcome {
+                            case_id: testcase.id(),
+                            exit_reason: "".to_string(),
+                            counterexample: testcase.value().clone(),
+                        })
+                    }
+                    _ => {
```

Comment:
Just in case, I would explicitly list the other two cases here in case we add more in the future.

Reply:
Yes, thanks for noticing. Changed everywhere